src/extension/byok/node/openAIEndpoint.ts

```diff
@@ -81,18 +79,26 @@ export class OpenAIEndpoint extends ChatEndpoint {
 		);
 	}
 
-	override createRequestBody(options: ICreateEndpointBodyOptions): IEndpointBody {
-		options.ignoreStatefulMarker = false;
-		const body = super.createRequestBody(options);
-		if (this.useResponsesApi) {
-			body.store = true;
-			body.n = undefined;
-			body.stream_options = undefined;
-			if (!this._modelInfo.capabilities.supports.thinking) {
-				body.reasoning = undefined;
+	protected override preprocessOptions(options: ICreateEndpointBodyOptions): ICreateEndpointBodyOptions {
+		return { ...options, ignoreStatefulMarker: false };
+	}
+
+	protected override getCapiCallback(): RawMessageConversionCallback | undefined {
```

Comment:
Both the callback and customizeCapiBody seems like you could handle it by overriding the main methods since they are working on data that is passed out of those methods? But all the endpoint stuff would be cleaner IMO if the completions/responses handling was separate from the endpoint. But I think that's a big refactoring

Reply:
Made it a bit better, we can do the detangling of endpoint and completions/responses later.